test/e2e/vsphere_volume_vsan_policy.go

```diff
@@ -0,0 +1,281 @@
+	// diskStripes value has to be between 1 and 12.
+	It("verify VSAN storage capability diskStripes - 14 and cacheReservation - 20 is not honored for dynamically provisioned pvc using storageclass", func() {
+		By("Invoking Test for VSAN policy diskStripes: 14, cacheReservation: 20")
+		scParameters[Policy_DiskStripes] = "14"
+		scParameters[Policy_CacheReservation] = "20"
+		framework.Logf("Invoking Test for VSAN storage capabilities: %+v", scParameters)
+		err := invokeInvalidVSANPolicyTestNeg(client, namespace, scParameters)
+		framework.Logf("Yo: %+q", err.Error())
+		errorMsg := "Invalid value for " + Policy_DiskStripes + " in volume plugin kubernetes.io/vsphere-volume."
+		if !strings.Contains(err.Error(), errorMsg) {
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+
+	// Invalid policy on a VSAN test bed.
+	// hostFailuresToTolerate value has to be between 0 and 3 including.
+	It("verify VSAN storage capability hostFailuresToTolerate - 4 is not honored for dynamically provisioned pvc using storageclass", func() {
+		By("Invoking Test for VSAN policy hostFailuresToTolerate: 4.")
+		scParameters[Policy_HostFailuresToTolerate] = "4"
+		framework.Logf("Invoking Test for VSAN storage capabilities: %+v", scParameters)
+		err := invokeInvalidVSANPolicyTestNeg(client, namespace, scParameters)
+		framework.Logf("Yo: %+q", err.Error())
+		errorMsg := "Invalid value for " + Policy_HostFailuresToTolerate + " in volume plugin kubernetes.io/vsphere-volume."
+		if !strings.Contains(err.Error(), errorMsg) {
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+
+	// Specify a valid VSAN policy on a non-VSAN test bed.
+	// The test should fail.
+	It("verify VSAN storage capability diskStripes - 1 and objectSpaceReservation - 30 and datastore - sharedVmfs-0 is not honored for dynamically provisioned pvc using storageclass", func() {
+		By("Invoking Test for VSAN policy diskStripes: 1, objectSpaceReservation: 30 and a non-VSAN datastore: sharedVmfs-0")
+		scParameters[Policy_DiskStripes] = "1"
+		scParameters[Policy_ObjectSpaceReservation] = "30"
+		scParameters[Datastore] = VmfsDatastore
+		framework.Logf("Invoking Test for VSAN storage capabilities: %+v", scParameters)
+		err := invokeInvalidVSANPolicyTestNeg(client, namespace, scParameters)
+		framework.Logf("Yo: %+q", err.Error())
+		errorMsg := "The specified datastore: \\\"" + VmfsDatastore + "\\\" is not a VSAN datastore. " +
+			"The policy parameters will work only with VSAN Datastore."
+		framework.Logf("errorMsg: %+q", errorMsg)
+		if !strings.Contains(err.Error(), errorMsg) {
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+})
+
+func invokeValidVSANPolicyTest(client clientset.Interface, namespace string, scParameters map[string]string) {
+	By("Creating Storage Class With VSAN policy params")
+	storageClassSpec := getVSphereStorageClassSpec("vsanpolicysc", scParameters)
+	storageclass, err := client.StorageV1beta1().StorageClasses().Create(storageClassSpec)
+	if err != nil {
```

Comment:
Do you really need these log statements? The expectation will print the error and the line number of which it occurred.